src/components/CheckIn/CheckInDetails.js

```diff
@@ -0,0 +1,238 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import { get } from 'lodash';
+
+import {
+  Callout,
+  Checkbox,
+  KeyValue,
+  Modal,
+  MultiColumnList,
+  Select,
+  TextField,
+} from '@folio/stripes/components';
+
+import getLocationsForSelect from '../Utils/getLocationsForSelect';
+import { fetchItems } from '../Receiving/util';
+import {
+  CHECKIN,
+  ITEMS,
+  LOCATIONS,
+} from '../Utils/resources';
+import { ITEM_STATUS } from '../Receiving/const';
+import CHECKIN_URLS from './const';
+import CheckInDetailsFooter from './CheckInDetailsFooter';
+import checkInItems from './util';
+
+import css from './CheckInDetails.css';
+
+class CheckInDetails extends Component {
+  static manifest = Object.freeze({
+    checkIn: CHECKIN,
+    items: ITEMS,
+    locations: LOCATIONS,
+    query: {},
+  })
+
+  static propTypes = {
+    close: PropTypes.func.isRequired,
+    location: PropTypes.object,
+    mutator: PropTypes.object,
+    items: PropTypes.arrayOf(PropTypes.object).isRequired,
+    resources: PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props);
+    this.callout = React.createRef();
+
+    this.state = {
+      isAllChecked: true,
+      isLoading: true,
+      items: this.props.items,
+    };
+  }
+
+  componentDidMount() {
+    const { mutator, items } = this.props;
```

Comment:
I think it's a lot of confusion here because of naming. I've called `items` array of object on Check-in Items screen to keep connection between what user see and what we show. In your case it's better to call a prop `pieces`, since you've also `items` from inventory.

Reply:
Updated